src/stan/callbacks/unique_stream_writer.hpp

```diff
@@ -28,7 +30,10 @@ class unique_stream_writer final : public writer {
    */
   explicit unique_stream_writer(std::unique_ptr<Stream>&& output,
                                 const std::string& comment_prefix = "")
-      : output_(std::move(output)), comment_prefix_(comment_prefix) {}
+      : output_(std::move(output)), comment_prefix_(comment_prefix) {
+    if (output_ == nullptr)
+      throw std::invalid_argument("output writer cannot be null");
+  }
```

Comment:
If we're preventing the output from being `nullptr` we can delete the a lot of the `if` statements that check for this in each operator() overload, such as on line 71

Reply:
fixed.